dev/deps/dependencies-client-flink-1.14

```diff
@@ -77,8 +77,8 @@ ratis-client/2.5.1//ratis-client-2.5.1.jar
 ratis-common/2.5.1//ratis-common-2.5.1.jar
 ratis-proto/2.5.1//ratis-proto-2.5.1.jar
 ratis-thirdparty-misc/1.0.4//ratis-thirdparty-misc-1.0.4.jar
-scala-library/2.12.15//scala-library-2.12.15.jar
-scala-reflect/2.12.15//scala-reflect-2.12.15.jar
+scala-library/2.12.18//scala-library-2.12.18.jar
+scala-reflect/2.12.18//scala-reflect-2.12.18.jar
```

Comment:
Why are we changing for flink 1.14, 1.15, etc ?
Should be only for spark 3.5, right ?

Reply:
> Why are we changing for flink 1.14, 1.15, etc ? Should be only for spark 3.5, right ?

https://github.com/apache/incubator-celeborn/pull/1154

Reply:
To make sure I am not missing something - these changes to flink are not required for this PR - but should have been part of 1154 ?

Reply:
@mridulm 
This PR migrate the default Scala version from 2.12.15 to 2.12.18 for service-related projects (master/worker) in order to comply with the minimum Scala version requirement of JDK 21, see https://docs.scala-lang.org/overviews/jdk-compatibility/overview.html. Additionally, the default Scala version is also utilized for compiling Flink-related projects.